sklearn/tests/test_common.py

```diff
@@ -60,21 +63,9 @@ def _tested_estimators():
             continue
         # FIXME _skip_test should be used here (if we could)
 
-        required_parameters = getattr(Estimator, "_required_parameters", [])
-        if len(required_parameters):
-            if required_parameters in (["estimator"], ["base_estimator"]):
-                if issubclass(Estimator, RegressorMixin):
-                    estimator = Estimator(Ridge())
-                else:
-                    estimator = Estimator(LinearDiscriminantAnalysis())
-            else:
-                warnings.warn("Can't instantiate estimator {} which requires "
-                              "parameters {}".format(name,
-                                                     required_parameters),
-                              SkipTestWarning)
-                continue
-        else:
-            estimator = Estimator()
+        with suppress(SkipTest):
+            estimator = _construct_instance(Estimator)
```

Comment:
So if this raises `SkipTest` we supress it and then return an `estimator` object (presumably from the previous iterations)?

Shouldn't that rather be,
```py
try:
    estimator = _construct_instance(Estimator)
except SkipTest:
    # possibly raise warning
    continue

yield name, estimator
```
or am I missing something?

Reply:
Good catch! I think you're right.

Reply:
is that fixed now? I am confused because the implementation looks like the thing you suggest

Reply:
This is fixed now.